js/app.jsx

```diff
@@ -297,101 +416,94 @@ var CaseverListItem = React.createClass({
+  }, render: function(){
+    var marker=""
+    if (this.state.order == ""){
+      marker = "▼";
+    }
+    else if (this.state.order == "-"){
+      marker = "▲";
+    }
+
+    return(
+      <th onClick={this.handleSort}>{this.props.text}{marker}</th>
+    )
+  }
+})
+
+var SearchTableHead = React.createClass({
   render: function() {
-    //can use the casevers.meta
-    var casevers = this.props.casevers.objects.map(function(casever){
-      return (<CaseverListItem casever={casever} onChange={this.props.handleCheck} handleAddFilter={this.props.handleAddFilter}/>)
-    }.bind(this))
+    var obj = this.props.setting.header;
+    var tableheader = obj.map(
+      function(setting){
+        if (setting["sortable"]) {
+          return <SearchTableHeadTh
+            text={setting["text"]}
+            filter={setting["filtername"]}
+            handleAddFilter={this.props.handleAddFilter} />
+        } else {
+          return <SearchTableHeadConstTh
+            text={setting["text"]} />
+        }
+      }.bind(this))
 
     return (
-      <Row>
-      <Table striped condensed hover className="caseverList">
-        <tbody>
-          <tr>
-            <th></th>
-            <th>ID</th>
-            <th>status</th>
-            <SortableTh name="name" filter="name" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <SortableTh name="priority" filter="case__priority" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <SortableTh name="product" filter="productversion" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <SortableTh name="modified" filter="modified_on" handleAddFilter={this.props.handleAddFilter}></SortableTh>
-            <th></th>
-            <th></th>
-          </tr>
-          {casevers}
-        </tbody>
-      </Table>
-      </Row>
+      <tr>{tableheader}</tr>
     )
   }
-});
-
-var SearchableCaseverList = React.createClass({
-  mixins: [SearchableRemoteListMixin],
-  api_url: config.baseUrl + "/api/v1/caseversion/",
-  //TODO: migrate to api_url: "https://moztrap.mozilla.org/api/v1/caseversionsearch/",
-  buildURL: function(query) {
-      return (buildQueryUrl(this.api_url, query, caseversionCodegen) + 
-              "&limit=" + config.defaultListLimit
-              //"&order_by=" + "-modified_on"
-             );
-  },
+})
 
-  handleQueueUpdate: function(e) {
-    if (e.target.checked){
-      var newState = {};
-      newState['checked'] = this.state['checked'].concat(e.target.value);
-      //console.log('will set state')
-      this.setState(newState);
-    }
-    else {
-      this.state['checked'].splice(this.state['checked'].indexOf(e.target.value), 1);
-      var newState = {};
-      newState['checked'] = this.state['checked'];
-      //console.log('will set state')
-      this.setState(newState);
-    }
-  },
-  /*
-  diff: function(){
-    alert(this.state.checked)
-  },
-  */
+var SearchTable = React.createClass({
```

Comment:
I guess we can drop the "Search" in this name, because without its parent and sibling node `<SearchFrom/>`, this class has no search involved. Maybe call it `MTTable` or somthing (MT=MozTrap) to avoid name collision.